includes/class-sensei-temporary-user-cleaner.php

```diff
@@ -84,37 +84,31 @@ public function clean_inactive_guest_users() {
 		$user_ids_to_be_deleted = $this->get_inactive_users();
 
 		foreach ( $user_ids_to_be_deleted as $user_id ) {
-			$course_ids = Sensei_Learner::instance()->get_enrolled_courses_query(
-				$user_id,
-				[
-					'posts_per_page' => -1,
-					'fields'         => 'ids',
-				]
-			)->posts;
-
-			foreach ( $course_ids as $course_id ) {
-				Sensei_Utils::sensei_remove_user_from_course( $course_id, $user_id );
-			}
-
-			Sensei_Temporary_User::delete_user( $user_id );
+			Sensei_Guest_User::delete_guest_user( $user_id );
+			Sensei_Preview_User::delete_preview_user( $user_id );
 		}
 	}
 
 	/**
-	 * Get a list of guest users who have not been active within last week.
+	 * Get a list of temporary users who have not been active within last week.
 	 *
 	 * @access private
 	 *
 	 * @return array List of user IDs.
 	 */
 	private function get_inactive_users() {
+
 		$guest_user_ids = get_users(
```

Comment:
Should we remove the filter by using `remove_filter( 'pre_user_query', [ Sensei_Temporary_User::class, 'filter_out_temporary_users' ], 11 );` before this call? Otherwise it'll probably not fetch the guest uesrs.

Reply:
Sorry about that, forgot to push the commit fixing that test. Should be green now!

Reply:
Thanks for updating @yscik ! But the cron wasn't still cleaning the users for me. I've checked a bit and found that the `get_users` function here isn't pulling any temporary users. Can you kindly check if it works for you?

For testing- 
- Take a couple of open access courses as Guest user
- Updated the wp_comments for them with old dates
- Install Wp Control plugin
- Find the `sensei_remove_inactive_guest_users` cron event and click on 'Run Now'